fluss-lake/fluss-lake-iceberg/src/test/java/com/apache/fluss/lake/iceberg/catalog/IcebergLakeCatalogTest.java

```diff
@@ -0,0 +1,334 @@
+        String database = "test_db";
+        String tableName = "multi_bucket_pk_table";
+
+        createNamespaceIfAbsent(database);
+
+        Schema flussSchema =
+                Schema.newBuilder()
+                        .column("user_id", DataTypes.BIGINT())
+                        .column("shop_id", DataTypes.BIGINT())
+                        .column("order_id", DataTypes.BIGINT())
+                        .column("amount", DataTypes.DOUBLE())
+                        .primaryKey("user_id", "shop_id") // Composite primary key
+                        .build();
+
+        TableDescriptor tableDescriptor =
+                TableDescriptor.builder()
+                        .schema(flussSchema)
+                        .distributedBy(4, "user_id", "shop_id") // Multiple bucket keys
+                        .build();
+
+        TableIdentifier tableId = TableIdentifier.of(database, tableName);
+
+        assertThatThrownBy(
+                        () -> {
+                            org.apache.iceberg.Schema icebergSchema =
+                                    flussIcebergCatalog.convertToIcebergSchema(tableDescriptor);
+                            PartitionSpec partitionSpec =
+                                    flussIcebergCatalog.createPartitionSpec(
+                                            tableDescriptor, icebergSchema);
+                            icebergCatalog.createTable(tableId, icebergSchema, partitionSpec);
+                        })
+                .isInstanceOf(UnsupportedOperationException.class)
+                .hasMessageContaining("Only one bucket key is supported for Iceberg");
+    }
+
+    @Test
+    void verifiesFlussCreatesValidIcebergTable() throws Exception {
+        String database = "test_db";
+
+        createNamespaceIfAbsent(database);
+
+        Schema flussSchema =
+                Schema.newBuilder()
+                        .column("user_id", DataTypes.BIGINT())
+                        .column("amount", DataTypes.DOUBLE())
+                        .primaryKey("user_id")
+                        .build();
+
+        TableDescriptor tableDescriptor =
+                TableDescriptor.builder().schema(flussSchema).distributedBy(4, "user_id").build();
+
+        // Create via Fluss iceberg catalog
+        TablePath flussTablePath = TablePath.of(database, "fluss_created");
+        flussIcebergCatalog.createTable(flussTablePath, tableDescriptor);
+
+        // Create equivalent via raw Iceberg catalog
+        org.apache.iceberg.Schema pureIcebergSchema =
+                new org.apache.iceberg.Schema(
+                        Types.NestedField.optional(1, "user_id", Types.LongType.get()),
+                        Types.NestedField.optional(2, "amount", Types.DoubleType.get()));
+
+        TableIdentifier pureTableId = TableIdentifier.of(database, "pure_iceberg");
+        Table pureIcebergTable = icebergCatalog.createTable(pureTableId, pureIcebergSchema);
```

Comment:
we don't care about iceberg's catalog can create table or not since it's iceberg implementation.